File: backend/src/models/Pipeline.ts
import {
    Table,
    Column,
    CreatedAt,
    UpdatedAt,
    Model,
    PrimaryKey,
    AutoIncrement,
    HasMany,
    ForeignKey,
    BelongsTo,
    Default,
    AllowNull,
    DataType
} from "sequelize-typescript";
import PipelineStage from "./PipelineStage";
import Tenant from "./Tenant";

@Table
class Pipeline extends Model<Pipeline> {
    @PrimaryKey
    @AutoIncrement
    @Column
    id: number;

    @Column
    name: string;

    @AllowNull(true)
    @Column(DataType.TEXT)
    description: string;

    @Default('kanban')
    @Column
    type: string; // 'kanban', 'funnel'

    @Default('#3B82F6')
    @Column
    color: string;

    @ForeignKey(() => Tenant)
    @Column(DataType.UUID)
    tenantId: number | string;

    @BelongsTo(() => Tenant)
    tenant: Tenant;

    @HasMany(() => PipelineStage)
    stages: PipelineStage[];

    @CreatedAt
    createdAt: Date;

    @UpdatedAt
    updatedAt: Date;
}

export default Pipeline;

File: bussines/internal/models/pipeline.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Pipeline struct {
	ID        int       `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	TenantID  uuid.UUID `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	CreatedAt time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt time.Time `gorm:"column:updatedAt" json:"updatedAt"`

	// Relations
	Stages []PipelineStage `gorm:"foreignKey:PipelineID" json:"stages,omitempty"`
}

func (Pipeline) TableName() string {
	return "Pipelines"
}

type PipelineStage struct {
	ID         int       `gorm:"primaryKey" json:"id"`
	Name       string    `gorm:"not null" json:"name"`
	PipelineID int       `gorm:"column:pipelineId;not null" json:"pipelineId"`
	Order      int       `gorm:"default:0" json:"order"`
	CreatedAt  time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt  time.Time `gorm:"column:updatedAt" json:"updatedAt"`
}

func (PipelineStage) TableName() string {
	return "PipelineStages"
}
